src/wp-includes/html-api/class-wp-html-processor.php

```diff
@@ -1752,7 +1752,19 @@ private function step_in_table(): bool {
 			 * > A character token, if the current node is table, tbody, template, tfoot, thead, or tr element
 			 */
 			case '#text':
-				return $this->step_in_table_text();
+				$current_token = $this->bookmarks[ $this->state->current_token->bookmark_name ];
+				if (
+					strspn( $this->html, "\u{0000}\u{0009}\u{000A}\u{000C}\u{000D}\u{0020}", $current_token->start, $current_token->length ) === $current_token->length
```

Comment:
note that there's no need for using the syntax here which has less support.

https://github.com/WordPress/wordpress-develop/pull/6977/files#diff-f822c7309a0a11d3993b378b85ef4a60588af0afac714f7e1590130aa4e300e5R1105

it's enough to use `\x00\x09\x0A\x0C\x0D\x20` which is I believe is identical to `\x00\t\r\f\n ` but the latter is far-more recognizable in code.

check out IN BODY as well where we take a different approach to this

https://github.com/WordPress/wordpress-develop/blob/339f70101549e7dd626495d6660f82872810a503/src/wp-includes/html-api/class-wp-html-processor.php#L1757-L1773

There's only a need to ignore the token if it's _only_ null bytes, otherwise we will rely on `get_modifiable_text()` to return the proper text.

Keep in mind that we should not be checking the raw HTML here. The spec is comparing already-decoded text, meaning that `&#x09;` is also ignored, since it decodes to U+09.

Reply:
This comment is on a stale commit, although some if the review is still valid. I'll update the characters used.

Reply:
> it's enough to use \x00\x09\x0A\x0C\x0D\x20 which is I believe is identical to \x00\t\r\f\n  but the latter is far-more recognizable in code.

I actually find the hex form easier to work with because I'm going back and forth with specifications that has definitions like this:

> ASCII whitespace is U+0009 TAB, U+000A LF, U+000C FF, U+000D CR, or U+0020 SPACE.

I'm not confident about the special escape sequences for all of these characters from memory.

How would you feel about adding a constant to one of the classes that can include documentation and can be reused in multiple places?

```php
/**
 * A string of the ASCII whitespace characters according to the Infra Standards.
 * 
 * > ASCII whitespace is U+0009 TAB, U+000A LF, U+000C FF, U+000D CR, or U+0020 SPACE.
 * @see https://infra.spec.whatwg.org/#ascii-whitespace
 * 
 * @var string
 */
const ASCII_WHITESPACE = /* I no longer care about this representation */;
```

Reply:
I'm reluctant to make sweeping changes at the moment. One problem with the constant is that many places in the spec add additional characters, so now on top of seeing the codes, you have to also correlate them with the additions, and if the additions are already contained within the constant.

If we feel strongly enough to use `\x09` instead of `\t` then we can address that in a comprehensive change. As it stands, everything uses the standard codes and I would prefer not introducing arbitrary change here.

Would be good to poll people and see if they recognize `"\x09\x0a\x0c\x0d\x20"` easier or `"\t\r\f\n "`

Reply:
I've changed the whitespace to use the more common escapes and updated the existing place to follow the ordering of whitespace characters in the specification. I also quoted that part of the spec in this case so there should be a clear correspondence and no confusion.